src/tmi/mod.rs

```diff
@@ -118,9 +158,50 @@ pub struct Chatters {
     pub viewers: Vec<Nickname>,
 }
 
+/// Possible options for a [TMIClient::get_hosts] request.
+#[derive(Debug)]
+pub enum HostsRequest {
+    /// Request a list of channels hosting a target broadcaster.
+    Target(UserId),
+    /// Request the broadcaster that a given channel is hosting.
+    Host(UserId),
+}
+
+/// Returned by TMI at `https://tmi.twitch.tv/hosts`
+///
+/// See [TMIClient::get_hosts]
+#[derive(Debug, Serialize, Deserialize)]
+pub struct GetHosts {
+    /// List of host records. `len()` will be 1 if requested for a [HostsRequest::Host],
+    /// in which case `target_id` may be missing if they are not hosting anyone.
+    pub hosts: Vec<Host>,
+}
+
+/// A host record returned by TMI at https://tmi.twitch.tv/hosts
+///
+/// See [TMIClient::get_hosts]
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Host {
+    /// User ID of the hosting channel
+    pub host_id: UserId,
+    /// User ID of the hosted channel. Will be missing if the given channel is not hosting anyone.
+    pub target_id: Option<UserId>,
+    /// Login of the hosting channel, if requested with `include_logins = true`
+    pub host_login: Option<Nickname>,
+    /// Login of the hosted channel, if requested with `include_logins = true`
+    pub target_login: Option<Nickname>,
+    /// Display name of the hosting channel, if requested with `include_logins = true`
+    pub host_display_name: Option<Nickname>,
+    /// Display name of the hosted channel, if requested with `include_logins = true`
+    pub target_display_name: Option<Nickname>,
+}
+
 /// Nickname
 pub type Nickname = String;
 
+/// User ID
+pub type UserId = u64; // TMI user ID's appear to still be ints, even though Helix uses strings.
```

Comment:
This is something I've been debating over, using ints or strings.

Might be worth to just get it over with and actually use ints in helix, can get that in for 0.5.0-alpha.3

Reply:
I actually started with this being `String`, but it failed to parse the response, so I switched it.

 Don't Helix User ID's actually get returned as strings, in quotes and everything? At least, that's what the examples in the [docs](https://dev.twitch.tv/docs/api/reference#get-users) show (not that the docs are always reliable). The requests can be made without the quotes, but the responses seem to always return strings.

Twitch4j actually switched *from* ints *to* strings for Helix *and* TMI, but Jackson seems to handle converting the ints to Strings automatically. Maybe there's a configuration option for `serde_json` to do the same?